### PR TITLE
EXL-300 Fixed printers view and hmarfa invoicer

### PR DESCRIPTION
**Pull request by zsso**
- Fixed printers view
- Fixed cartiridges view
- Fixed contracts view
- Fixed reminders view
- Added missing rigths
- Fixes hmarfa invoicer
- Added additional table indexes to glpi_plugin_fields_cartridgecartridgecustomfields

### CODE REVIEW

File: plugin/front/hmarfaexport.form.php [hunk 1]
```diff
@@ -31,4 +33,3 @@
 
 PluginIserviceHmarfa::showExportForm($id, $mode);
 
-Html::footer();
```
Comment (hupupu) [inline plugin/front/hmarfaexport.form.php:34]: why don't we need footer?

File: plugin/inc/specialviews/reminders.class.php [hunk 2]
```diff
@@ -77,7 +78,7 @@ protected function getSettings(): array
 							[user_id]
 						",
             'default_limit' => 50,
-            'row_class' => 'function:PluginIserviceView_Reminders::getRowBackgroundClass($row);',
+            'row_class' => 'function:\GlpiPlugin\Iservice\Specialviews\Reminders::getRowBackgroundClass($row_data);',
```
Comment (hupupu) [inline plugin/inc/specialviews/reminders.class.php:81]: sometimes it is `row`, sometimes it is `row_data`. research.

Comment (zsso) [reply plugin/inc/specialviews/reminders.class.php:81]: In this file sometimes `row` sometimes `row_data` is needed.

File: plugin/front/hmarfaexport.form.php (hunk 1 above)
Comment (zsso) [reply plugin/front/hmarfaexport.form.php]: Fixed

File: plugin/inc/specialviews/printers.class.php [hunk 3]
```diff
@@ -435,7 +436,7 @@ protected function getSettings(): array
                             , s.cm_field
                             , t2.codbenef
                             , t2.numar_facturi_neplatite
-                        FROM glpi_plugin_iservice_printers p
+                        FROM glpi_plugin_iservice_printers_basic p
```
Comment (hupupu) [inline plugin/inc/specialviews/printers.class.php:439]: leave the `glpi_plugin_iservice_printers`

File: plugin/install/sql/create_views.sql [hunk 4]
```diff
@@ -80,6 +147,57 @@ from (((`glpi_printers` `p`
     left join `glpi_plugin_fields_printerprintercustomfields` cfp on cfp.items_id = p.id and cfp.itemtype = 'Printer'
     left join glpi_plugin_iservice_printers_last_closed_tickets plct on plct.printers_id = p.id;
 
+create or replace view glpi_plugin_iservice_printers_basic as
```
Comment (hupupu) [inline plugin/install/sql/create_views.sql:150]: the original must be `glpi_plugin_iservice_printers` (with every join except last closed tickets) and the other to `glpi_plugin_iservice_printers_with_last_closed_ticket_data`

**Review by hupupu — DISMISSED** [review]

**Review by hupupu — APPROVED** [review]

**Pull request merged**